### PR TITLE
Proxy authorization header https only

### PR DESCRIPTION
**Pull request by rebloor**
#### Summary
Adds documentation that `proxyAuthorization` and `connectionIsolation` in `proxy.onRequest` only apply to HTTPS proxies. This covers the documentation requirements for the original bug [Bug 1549368](https://bugzilla.mozilla.org/show_bug.cgi?id=1549368) _Limit proxy.onRequest to set proxyAuthorization and connectionIsolation to apply only to https proxies_  and subsequent [Bug 1618494](https://bugzilla.mozilla.org/show_bug.cgi?id=1618494) _Returning proxyAuthorizationHeader from proxy.onRequest does not actually send Proxy-Authorization header_.

#### Metadata

This PR…

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

**Review by mixedpuppy — APPROVED** [review]

File: files/en-us/mozilla/add-ons/webextensions/api/proxy/proxyinfo/index.md [hunk 1]
```diff
@@ -42,12 +42,12 @@ Values of this type are objects. They contain the following properties:
...
 - `proxyAuthorizationHeader`
 
-  - : `string.` This string, if set to non-empty, is passed directly as a value to the {{httpheader("Proxy-Authorization")}} request header sent to HTTP proxies as part of plain HTTP requests and CONNECT requests. In other words, this can be used to directly authenticate to HTTP proxies requiring (non-challenging) authentication.
+  - : `string.` This string, if set to non-empty, is passed directly as a value to the {{httpheader("Proxy-Authorization")}} request header sent to HTTPS proxies as part of HTTPS and CONNECT requests. In other words, this can be used to directly authenticate to HTTPS proxies that allow (non-challenging) authentication.
 
     For instance, if you want to send "username" and "password" for "basic" authentication, you can set the `proxyAuthorizationHeader` property to `Basic dXNlcm5hbWU6cGFzc3dvcmQ=`
 
 - `connectionIsolationKey` {{optional_inline}}
-  - : `string.` An optional key used for additional isolation of this proxy connection.
+  - : `string.` An optional key used for additional isolation of this proxy connection. Applies to HTTPS proxies only.
```
Comment (Rob--W) [inline files/en-us/mozilla/add-ons/webextensions/api/proxy/proxyinfo/index.md:50]: @rebloor Could you revert this change? It is not specific to https only, according to my findings at https://bugzilla.mozilla.org/show_bug.cgi?id=1794464#c14

Comment (rebloor) [reply files/en-us/mozilla/add-ons/webextensions/api/proxy/proxyinfo/index.md:50]: Done, see https://github.com/mdn/content/pull/33950
